lxd/db/instances.go

```diff
@@ -159,6 +159,15 @@ type InstanceArgs struct {
 	ExpiryDate   time.Time
 }
 
+// InstanceTypeFilter returns an InstanceFilter populated with a valid instance type,
+// or an empty filter if instance type is 'Any'.
+func InstanceTypeFilter(instanceType instancetype.Type) InstanceFilter {
```

Comment:
@masnax not for this PR, but another item on your work list, is I think, to remove `instance.Any` entirely and ensure that everywhere we accept an `instancetype.Type` we accept it as a pointer, where nil supplied means "any", to align with the DB filtering now thats changed.